libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/QnAMaker/QnAMakerRecognizer.cs

```diff
@@ -96,6 +96,30 @@ public QnAMakerRecognizer()
         [JsonProperty("rankerType")]
         public StringExpression RankerType { get; set; } = RankerTypes.DefaultRankerType;
 
+        /// <summary>
+        /// Gets or sets the whether to include the dialog name metadata for QnA context.
+        /// </summary>
+        /// <value>
+        /// A bool or boolean expression.
+        /// </value>
+        [DefaultValue(true)]
+        [JsonProperty("includeDialogNameInMetaData")]
+        public BoolExpression IncludeDialogNameInMetaData { get; set; } = true;
```

Comment:
Casing: `Metadata` is a word on its own, so casing should be IncludeDialogNameInMetadata